homeassistant/components/firmata/pin.py

```diff
@@ -151,3 +202,66 @@ async def latch_callback(self, data: list) -> None:
             return
         self._state = new_state
         self._forward_callback()
+
+
+class FirmataAnalogInput(FirmataBoardPin):
+    """Representation of a Firmata Analog Input Pin."""
+
+    def __init__(
+        self, board: FirmataBoard, pin: FirmataPinType, pin_mode: str, differential: int
+    ):
+        """Initialize the analog input pin."""
+        self._differential = differential
+        self._forward_callback = None
+        super().__init__(board, pin, pin_mode)
+
+    async def start_pin(self, forward_callback: Callable[[], None]) -> None:
+        """Get initial state and start reporting a pin."""
+        _LOGGER.debug(
+            "Starting reporting updates for analog input pin %s on board %s",
+            self._pin,
+            self.board.name,
+        )
+        self._forward_callback = forward_callback
+        api = self.board.api
+        # Only PIN_MODE_ANALOG_INPUT mode is supported as sensor input
+        await api.set_pin_mode_analog_input(
+            self._analog_pin, self.latch_callback, self._differential
+        )
+
+        self._state = (await self.board.api.analog_read(self._analog_pin))[0]
+
+        self._forward_callback()
+
+    async def stop_pin(self) -> None:
+        """Stop reporting analog input pin."""
+        _LOGGER.debug(
+            "Stopping reporting updates for analog input pin %s on board %s",
+            self._pin,
+            self.board.name,
+        )
+        api = self.board.api
+        await api.disable_analog_reporting(self._analog_pin)
+
+    @property
+    def state(self) -> int:
+        """Return sensor state."""
+        return self._state
+
+    @callback
```

Comment:
A coroutine can't be a callback.

Reply:
I will remove from `FirmataBinaryDigitalInput` as well.